[Jerome Soh] iP

Jeromesyl opened the pull request:
# DukePro
> At your service, my lord. - Retainer

### DukePro is at your service, meeting all your task-ly needs. It is,

- text-based
- easy to learn
- fast to use

### All you need to do is,

1. download it from [here](https://github.com/Jeromesyl/ip/releases/tag/A-Jar).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you.

Of course, it is **FREE!**
![gold hmm](https://user-images.githubusercontent.com/60766419/131476016-507f4dcf-439a-428a-8b3e-3ecd2db901e9.PNG)

### Features:
- [x]  Managing todos
- [x] Managing deadlines 
- [x] Managing events
- [x] Find tasks on a specific date
- [ ] Reminders (coming soon)

If you're a java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Jeromesyl committed:
Add user input and program exit

Jeromesyl committed:
Add level 2. Add, List

Jeromesyl committed:
Add Level 3: Mark as done

Jeromesyl committed:
Add Level 4. Todo, Event, Deadline

Jeromesyl committed:
Add runtest file, Update output format

Jeromesyl committed:
Add Level 5. Handle errors

Jeromesyl committed:
Add Level.6 Delete

Jeromesyl committed:
Document javaDocs for classes and methods

Jeromesyl committed:
Add Level 7. Save

Jeromesyl committed:
Add Level-8 Date/Time

Jeromesyl committed:
Update loadfile method

Jeromesyl committed:
Update date time format

Jeromesyl committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Duke.java

Jeromesyl committed:
Add moreOOP

Jeromesyl committed:
Move files into packages

Jeromesyl committed:
Add JUnit tests

Jeromesyl committed:
Create jar file

Jeromesyl committed:
Revert "Create jar file"

This reverts commit 539f065a37c7fa4e88fac2ed60e47bf433dec6ec.

Jeromesyl committed:
Add JavaDocs

Jeromesyl committed:
Update coding standard

Jeromesyl committed:
Add Level.8 FindCommand

Jeromesyl committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/task/Todo.java

Jeromesyl committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/TaskList.java
#	src/main/java/duke/Ui.java

Jeromesyl committed:
Add java docs

Jeromesyl committed:
Merge branch 'add-gradle-support' of https://github.com/Jeromesyl/ip into branch-A-gradle

Jeromesyl committed:
Set up Gradle support and Checkstyle plugin

Jeromesyl committed:
Implement GUI

1waykiat commented on src/test/java/duke/TaskListTest.java:
```diff
@@ -0,0 +1,37 @@
+
+
+
+public class TaskListTest {
```
Perhaps could test for cases where an Exception should occur?

1waykiat reviewed (commented):
Overall the code looks really neat without any major issues. 👍

Jeromesyl committed:
Implement duke with GUI

chetwinlow commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,78 @@
+    }
+
+    /**
+     * You should have your own function to generate a response to user input.
```
Maybe you could start your comment on this method with a Verb? Example for this would be 
/**
 Returns a response ....

chetwinlow commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,147 @@
+    }
+
+    /**
+     * Print the tasks that match the filter.
```
Probably careless but you probably meant to type "Prints"?

chetwinlow commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,55 @@
+     * @return String to write to duke.txt
+     */
+    @Override
+    public String toWrite() {
```
I do feel that this method name is a little ambiguous or might be misinterpreted? Might I suggest something like "convertToWriteFormat" or "toWriteFormat"

Jeromesyl replied:
Thanks for the suggestion!

ryanpeh commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,147 @@
+     */
+    public String taskAdded(Task task) {
+        return ("Okay! I've added this task:\n    " + task.toString()) + "\n";
+    }
```
`taskAdded` is a noun, perhaps you could change it to `getTaskAdded` or prefix with any other action verb?

ryanpeh commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,147 @@
+     */
+    public String taskMarked(Task task) {
+        return ("Okay! This task has been marked done:\n  " + task.toString() + "\n");
+    }
```
Similar to before, perhaps it should be `getTaskMarked` or something similar instead?

ryanpeh commented on src/main/java/duke/command/ExitCommand.java:
```diff
@@ -0,0 +1,36 @@
+
+    /**
+     * Indicates if the command ends the program after executing.
+     * @return true
```
Perhaps there should be a space between the description and the block tag?

ryanpeh commented on src/main/java/duke/command/AddCommand.java:
```diff
@@ -0,0 +1,86 @@
+public class AddCommand extends Command {
+    private Task task;
+
+    /**
+     * Class constructor.
+     * @param input The user input to indicate the task to add.
+     * @throws DukeException If the command format is invalid.
+     */
```
Perhaps there should be a space between the description and the block tags?

ryanpeh commented on src/main/java/duke/command/AddCommand.java:
```diff
@@ -0,0 +1,86 @@
+    /**
+     * Indicates if the command ends the program after executing.
+     * @return false
+     */
```
Perhaps there should be a space between the description and the block tag?

ryanpeh commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,147 @@
+     */
+    public String taskAdded(Task task) {
+        return ("Okay! I've added this task:\n    " + task.toString()) + "\n";
+    }
+
+    /**
+     * Prints the current number of tasks in the TaskList.
+     *
+     * @param taskList The TaskList object containing all tasks.
+     */
+    public String showTaskListSize(TaskList taskList) {
+        return String.format("You have %d task(s) in the list.\n\n", taskList.getSize());
+    }
+
+    /**
+     * Prints the task that was successfully deleted.
+     *
+     * @param task The task object that was deleted.
+     */
+    public String taskDeleted(Task task) {
+        return ("Okay! I've removed this task:\n    " + task.toString());
+    }
+
+    /**
+     * Prints the task that was marked as done.
+     *
+     * @param task The task object that was marked done.
+     */
+    public String taskMarked(Task task) {
+        return ("Okay! This task has been marked done:\n  " + task.toString() + "\n");
+    }
+
+    /**
+     * Prints the goodbye message.
+     */
+    public String showGoodbye() {
+        scanner.close();
+        return "Bye bye!\n";
+    }
+}
```
Perhaps you should update your javadocs from `Prints` to `Returns` since it does not print anything anymore.

Jeromesyl replied:
Thanks! I've updated the code violations.

Jeromesyl committed:
Update JavaDocs and coding style violations

Jeromesyl committed:
Add assertions and update coding standard violations

Certain parts of the code must have assumptions that can hold.

Some JavaDocs and code were updated to look neater after implementing
GUI. Some method names needed to be changed to be more concise.

Jeromesyl committed:
Update method names and improve code quality.

Some method names were changed to make it more concise and to improve
code quality.

JavaDocs are further updated to reflect the proper parameters and return
values.

Jeromesyl committed:
Merge pull request #2 from Jeromesyl/branch-A-CodeQuality

Update method names and improve code quality.

Jeromesyl committed:
Merge branch 'master' into branch-A-Assertions

Jeromesyl committed:
Merge pull request #1 from Jeromesyl/branch-A-Assertions

Add assertions and update coding standard violations

Jeromesyl committed:
Implement method to detect duplicate tasks

Duplicate tasks can be added to the task list, which may crowd the list
and make it hard to add future methods such as sorting.

It is necessary to differentiate between task objects so as to prevent
complications in the future.

Jeromesyl committed:
Add Help Command

The usage menu only appears when the app launches. Users need to refer
to the commands sometimes, so having a command to show the usage menu
will be helpful.

Jeromesyl committed:
Updated GUI

The app used to launch with a blank dialog box. It's good to have a
welcome message to introduce the user to the app.

GUI changes:
- Text is now centered to the image with some padding.
- Errors are shown differently, with an added "ERROR" in front.
- Background color of the user and Chad dialog boxes are different to
better distinguish between them

Jeromesyl committed:
Add Ui screenshot to docs.

Jeromesyl committed:
Add user guide for chadbot

Jeromesyl committed:
Set theme jekyll-theme-slate

Jeromesyl committed:
Small fix for Launcher.

Jeromesyl committed:
Merge branch 'master' of https://github.com/Jeromesyl/ip
